sds/src/scanner/mod.rs

```diff
@@ -1206,4 +1206,25 @@ mod test {
         assert_eq!(matches.len(), 1);
         assert_eq!(content, "[5184[credit card]]".to_string());
     }
+
+    #[test]
+    fn test_internal_overlapping_matches_multibyte_char() {
+        // simple "credit-card" rule
+        let rule_0 = RuleConfig::builder("(\\d+){4}(,\\D+){4}".to_owned())
+            .match_action(MatchAction::Redact {
+                replacement: "[credit card]".to_string(),
+            })
+            .validator(LuhnChecksum)
+            .build();
+
+        let scanner = Scanner::new(&[rule_0]).unwrap();
+
+        // The first 4 numbers match as a credit-card, but fail the luhn checksum.
+        // The last 4 numbers (which overlap with the first match) pass the checksum.
+        let mut content = "[5184,5185,5252,5052€5005]".to_string();
```

Comment:
maybe let's add `€` in the regex. Also, the regex could me the same as previous one `([,€]?\\d+){4}`